flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/debezium/task/MySqlSnapshotSplitReadTask.java

```diff
@@ -313,7 +316,14 @@ else if (actualColumn.jdbcType() == Types.TINYINT
         else if ("CHAR".equals(actualColumn.typeName())
                 || "VARCHAR".equals(actualColumn.typeName())
                 || "TEXT".equals(actualColumn.typeName())) {
-            return rs.getBytes(fieldNo);
+            Charset columnCharset = DebeziumUtils.charsetFor(actualColumn);
+            final byte[] stringColumnBytes = rs.getBytes(fieldNo);
```

Comment:
After investigate, I find `return rs.getString()` is enough.

Reply:
It seems that the ResultSetImpl.getString will convert the result with the right configuration. But I read this comment:
```
// DBZ-2673
        // It is necessary to check the type names as types like ENUM and SET are
        // also reported as JDBC type char
        else if ("CHAR".equals(actualColumn.typeName())
                || "VARCHAR".equals(actualColumn.typeName())
                || "TEXT".equals(actualColumn.typeName())) {
            return rs.getString(fieldNo);
        }
```
Will some other bugs occur by this change for the types like ENUM and SET ?

Reply:
DBZ-2673[1] is not about the SET or ENUM type and it is used to fix the charset problems. DBZ-2673 wants to get the raw bytes for the column with latin1 charset. It made changes:

1. override the jdbc url to set characterSetResults null(utf-8 by default)
2. get the rawbytes using rs.getBytes.

However, if the characterSetResults is utf-8, the rs.getBytes is equal to the rs.getString().getBytes("UTF-8"), which causes the problem here.

After read the fix in the DBZ-3700, I think it's on the right way to solve all the problem here[2,3]. And add more tests to verify.

[1] https://issues.redhat.com/browse/DBZ-2673
[2] https://issues.redhat.com/browse/DBZ-3700
[3] https://github.com/debezium/debezium/pull/3128/commits/6ee9ffca073be66ecce31712789672ea6df0f5c4

Reply:
I refactor the code with the DBZ-3700.
This api can not be found : connectorConfig.customConverterRegistry().isEmpty(). Do I need to revert the change in commit 7f79cb9 ?